source/Databricks/source/SqlStatementExecution.UnitTests/Internal/DatabricksSqlStatementClientTests.cs

```diff
@@ -74,6 +74,31 @@ public DatabricksSqlStatementClientTests()
             ExecuteAsync("some sql", new List<SqlStatementParameter>()).ToListAsync());
     }
 
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task StreamAsync_WhenFirstRunningThenStatementFails_ThrowsDatabricksSqlException(
+        Guid statementId,
+        Mock<ISqlResponseParser> parserMock,
+        DatabricksSqlStatementClientBuilder builder)
+    {
+        // Arrange
+        parserMock
+            .Setup(parser => parser.ParseStatusResponse(_running))
+            .Returns(SqlResponse.CreateAsRunning(statementId));
+        parserMock
+            .Setup(parser => parser.ParseStatusResponse(_failed))
+            .Returns(SqlResponse.CreateAsFailed(statementId));
+        var sut = builder
+            .AddHttpClientResponse(_running)
+            .AddHttpClientResponse(_failed)
+            .UseParser(parserMock.Object)
+            .Build();
+
+        // Act and assert
+        await Assert.ThrowsAsync<DatabricksSqlException>(async () => await sut.
+            StreamAsync("some sql", new List<SqlStatementParameter>()).ToListAsync());
```

Comment:
Consider using FluentAssertions and split "act" from "assert" like:

```suggestion
        // Act
        Func<Task> act = async () => await sut.ExecuteAsync("some sql", new List<SqlStatementParameter>()).ToListAsync();

        // Assert
        await act.Should().ThrowAsync<DatabricksSqlException>();
```